src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -217,6 +217,8 @@ private Task InvokeAllResourceFiltersAsync()
             _cursor.Reset();
 
             _resourceExecutingContext = new ResourceExecutingContext(_controllerContext, _filters);
+            _resourceExecutingContext.ValueProviderFactories = _controllerContext.ValueProviderFactories;
```

Comment:
Can you add a unit test that shows that modifying the list of VPFs in a resource filter also modifies the list of VPFs on `controller.ControllerContext`?